### ISSUE TITLE
[BUG] `def` function will output string with single quotation marks, but `fn` fucntion will not.  

### ISSUE THREAD

**Issue by jimax2000** [modularml/mojo#2207]
### Bug description

The same functions output different string result when using `def` and `fn` ,  `def` function will output string with single quotation marks, but `fn` fucntion will not.  

### Steps to reproduce

Here is a simple mojo file, use `def` and `fn` :
```
def sayHello(name):
        return "Hello, " + name + "!"

fn sayHello2(name: String) -> String:
        return "Hello, " + name + "!"

def main() :
        ret1 = sayHello("Alice")
        print("sayHello:", ret1)

        ret2 = sayHello2("Bob")
        print("sayHello2:", ret2)

        s = ret1 + ret2
        print(s)
```
It will output:
```
sayHello: 'Hello, Alice!'
sayHello2: Hello, Bob!
'Hello, Alice!'Hello, Bob!
```

### System information

```shell
- What OS did you do install Mojo on ? 
Ubuntu 22.04
- Provide version information for Mojo by pasting the output of `mojo -v`
mojo 24.2.0 (c2427bc5)
- Provide Modular CLI version by pasting the output of `modular -v`
modular 0.6.0 (04c05243)
```

**Comment by past-moon** [comment]
`def` functions return a `object` by default, and `object.__str__` explicitly adds quotes to the string:

https://github.com/modularml/mojo/blob/f1493c87ff8cbabb3cb88fb11fd1063403a7ffe2/stdlib/src/builtin/object.mojo#L604-L613

So I'm not sure if this counts as a bug or intended behavior.

Now what is odd is when you write a function like this: `def sayHello(name) -> String:` it still puts the quotes around it. Maybe despite the return type it is still going through `object` first?

**Comment by gabrieldemarmiesse** [comment]
I think that we have here a small issue in the stdlib. We use only `__str__`, while in python there is `__str__` and `__repr__`. Notably it has an impact when implementing `List.__str__` and `Dict.__str__`.

In python, we have this:

```python
>>> print(("hello").__str__())
hello
>>> print(("hello").__repr__())
'hello'
>>> print(("hel'lo").__str__())
hel'lo
>>> print(("hel'lo").__repr__())
"hel'lo"
```
Notice how the quotes are different? By using only `__str__` in Mojo, we can't represent well the quotes. This notably leads to those issues:

```mojo
a = List[String]("hello", "world")
print(__type_of(a).__str__(a))
# [hello, world]
```
In python this would print `['hello', 'world']`.
I'll see if I can make a PR to introduce this dunder method

**Comment by JoeLoser** [comment]
I agree.  This is "known not implemented yet" from my perspective (but no public GitHub issues exist or anything of the like unfortunately).  When I originally implemented `Stringable` trait, `str(...)` and the like, this was knowingly not worried about at the time.  I do think we should implement a meaningful `__repr__` pretty much for all types in the standard library.  At the time, it just wasn't a priority.  😄 

FYI @ConnorGray in case you have thoughts here.

@walter-erquinigo @River707 for visibility with regard to what our debugger should use.

**Comment by walter-erquinigo** [comment]
I think that eventually we should allow __repr__ to be used as a fallback visualizer, but not __str__. We'll get there eventually once we start working on expression evaluation.

**Comment by ConnorGray** [comment]
I don't have much to add, I agree with the argument and the context Joe added. This is something we should have, but haven't gotten to yet.

**Comment by conorbronsdon** [comment]
Good news! As of [v26.2](https://docs.modular.com/mojo/changelog/#v0262-2026-03-19), `def` and `fn` have identical behavior, so any behavioral differences are resolved. Closing with that in mind - thanks for flagging!
